firebase-installations/src/main/java/com/google/firebase/installations/StateListener.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations;
+
+import com.google.android.gms.tasks.TaskCompletionSource;
+import com.google.firebase.installations.local.PersistedFidEntry;
+
+interface StateListener {
+  boolean onStateReached(PersistedFidEntry persistedFidEntry);
+}
+
+class GetIdListener implements StateListener {
```

Comment:
Don't include multiple classes in a single file. This file should have one class / interface (StateListener) and the others should either be inner classes or separate files.

Reply:
https://google.github.io/styleguide/javaguide.html#s3.4.1-one-top-level-class

Reply:
Thanks for the link. I thought about it but wasn't sure what the guidelines was. I will keep in mind for future changes.